JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -460,77 +420,77 @@ private fun InterestsTabRowContent(
 }
 
 /**
- * Return the interest sections grouped given the number of columns that fill the screen.
- * Sections are distributed to balance the number of items on each column.
- */
-@Composable
-private fun rememberSectionsGroupedInColumns(
-    sections: List<InterestSection>,
-    columns: Int
-): List<List<InterestSection>> = remember(sections, columns) {
-    val sectionsInColumns = List<MutableList<InterestSection>>(columns) { mutableListOf() }
-    val elementsInColumns = IntArray(columns)
-    // For each section, find the column with less elements and add it there
-    sections.forEach { section ->
-        val shortestColumnIndex = elementsInColumns.withIndex().minByOrNull { it.value }!!.index
-        sectionsInColumns[shortestColumnIndex] += section
-        elementsInColumns[shortestColumnIndex] += section.interests.size
-    }
-    sectionsInColumns.toList()
-}
-
-/**
- * Returns the max width for a Topic Item given maxWidth and number of columns constraints.
+ * Custom layout for the Interests screen that places items on the screen given the available size.
+ *
+ * For example: Given a list of items (A, B, C, D, E) and a screen size that allows 2 columns,
+ * the items will be displayed on the screen as follows:
+ *     A B
+ *     C D
+ *     E
  */
 @Composable
-private fun rememberItemMaxWidth(windowMaxWidth: Dp, columns: Int) =
-    remember(windowMaxWidth, columns) {
-        if (columns == 1) {
-            // If the space available is small, fill the screen
-            if (windowMaxWidth < 600.dp) Dp.Infinity else 600.dp
+private fun InterestsAdaptiveContentLayout(
+    modifier: Modifier = Modifier,
+    topPadding: Dp = 0.dp,
+    itemSpacing: Dp = 8.dp,
+    itemMaxWidth: Dp = 450.dp,
+    content: @Composable () -> Unit,
+) {
+    val oneColumnWidthPx = with(LocalDensity.current) { 600.dp.roundToPx() }
+    val topPaddingPx = with(LocalDensity.current) { topPadding.roundToPx() }
+    val itemSpacingPx = with(LocalDensity.current) { itemSpacing.roundToPx() }
+    val itemMaxWidthPx = with(LocalDensity.current) { itemMaxWidth.roundToPx() }
+
+    Layout(modifier = modifier, content = content) { measurables, outerConstraints ->
+        // Number of columns to display on the screen
+        val columns = if (outerConstraints.maxWidth < oneColumnWidthPx) 1 else 2
```

Comment:
It could and the rest of the code is flexible enough to respond to any number of columns. However, the design mocks only have two columns max, added a comment to clarify this